opencog/nlp/relex2logic/rule-helpers.scm

```diff
@@ -414,30 +414,35 @@
 )
 ; -----------------------------------------------------------------------
 ; Adjective and adverb rules
+; (advmod is also called for adverbials, such as some prepositional phrases)
+;
+; Prepositional adverbial example: "The man sang with a powerful voice."
+; Prepositional adjectival example: "The man with a powerful voice sang."
+;
 ; -----------------------------------------------------------------------
 (define (amod-rule concept instance adj adj_instance)
 	(list (InheritanceLink  (ConceptNode adj_instance df-node-stv) (ConceptNode adj df-node-stv) df-link-stv)
-	(InheritanceLink  (ConceptNode instance df-node-stv) (ConceptNode adj_instance df-node-stv) df-link-stv)
 	(InheritanceLink  (ConceptNode instance df-node-stv) (ConceptNode concept df-node-stv) df-link-stv)
-	)
+	(InheritanceLink  (ConceptNode instance df-node-stv) (ConceptNode adj_instance df-node-stv) df-link-stv))
 )
 
 (define (advmod-rule verb instance adv adv_instance)
 	(list (InheritanceLink  (ConceptNode adv_instance df-node-stv) (ConceptNode adv df-node-stv) df-link-stv)
-	(InheritanceLink  (SatisfyingSetLink (PredicateNode instance df-node-stv) df-link-stv) (ConceptNode adv_instance df-node-stv) df-link-stv)
 	(ImplicationLink  (PredicateNode instance df-node-stv) (PredicateNode verb df-node-stv) df-link-stv)
+	(InheritanceLink  (SatisfyingSetLink (PredicateNode instance df-node-stv) df-link-stv) (ConceptNode adv_instance df-node-stv) df-link-stv)
 	)
 )
-
-;(define (prepadvmod-rule verb_concept verb_instance noun_concept noun_instance prep_concept prep_instance)
-;	(list 
-;		(InheritanceLink (ConceptNode noun_instance df-node-stv) (ConceptNode noun_concept df-node-stv) df-link-stv)
-;		(InheritanceLink (ConceptNode prep_instance df-node-stv) (ConceptNode prep_concept df-node-stv) df-link-stv)
-;		
-;		(ImplicationLink (PredicateNode verb_instance df-node-stv) (PredicateNode verb_concept df-node-stv) df-link-stv)
-;		(InheritanceLink (SatisfyingSetLink (PredicateNode instance df-node-stv) df-link-stv) (ConceptNode adv_instance df-node-stv) df-link-stv)
-;	)
-;)
+;-----------------------------------------------------------------------
+; prepositional phrase rule
+;-----------------------------------------------------------------------
+(define (pp-rule prep_concept prep_instance noun_concept noun_instance)
+	(list (ImplicationLink (PredicateNode prep_instance df-node-stv) (PredicateNode prep_concept df-node-stv) df-link-stv)
+	(InheritanceLink (ConceptNode noun_instance df-node-stv) (ConceptNode noun_concept df-node-stv) df-link-stv)
+	(EvaluationLink df-link-stv
+		(PredicateNode prep_instance df-node-stv)
+		(ListLink (ConceptNode noun_instance df-node-stv))
+	))
+)
```

Comment:
The output is this pp-rule seems incomplete.  For example, it will generate
```
(EvaluationLink
  (PredicateNode "on@11")
  (ListLink
    (ConceptNode "table@22")
  )
)
```
without saying what is "on-table".  However, I noticed you are using the "SVO-rule" to generate the full link (it-on-table) as well, so maybe the pp-rule is not needed.

Probably should rename SVO-rule now that it is used for so many purpose...